src/ui/components/atoms/TitleSubtitle/TitleSubtitle.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import type { IProps } from './types/IProps';
+import styles from './TitleSubtitle.module.scss';
+
+function TitleSubtitle({
+  title,
+  subtitle,
+  size = 'md',
+  alignment = 'left',
+  variant = 'default',
+  titleTag = 'h2',
+  subtitleTag = 'p',
+  className = '',
+}: IProps) {
+  const containerClass = `${styles['title-subtitle']} 
+    ${styles[`title-subtitle--size-${size}`]} 
+    ${styles[`title-subtitle--alignment-${alignment}`]} 
+    ${styles[`title-subtitle--variant-${variant}`]} 
+    ${className}`.trim();
+
+  const titleClass = `${styles['title-subtitle__title']} ${
+    styles[`title-subtitle__title--${titleTag}`]
+  }`;
+  const subtitleClass = `${styles['title-subtitle__subtitle']} ${
+    styles[`title-subtitle__subtitle--${subtitleTag}`]
+  }`;
+
+  const TitleTag = titleTag as keyof JSX.IntrinsicElements;
```

Comment:
i don't agree with this approach the title tag should be a separated component from this, please put this code in another place, And please use something more clarify like 
```
function titleTag ({children} : React.node) {
 return <h2> {children} </h2>
}
```